### PR TITLE
Upgrade deps and remove CRA preset from Storybook

### PR DESCRIPTION
**Pull request by axelboc**
I ran into issues upgrading Storybook, so I decided to try removing CRA again and I finally got there.

Upgrading Three.js still breaks the shader-based visualizations: `texture2D` seems to always return `0`.

### CODE REVIEW

File: apps/storybook/.storybook/main.js [hunk 1]
```diff
@@ -6,23 +6,8 @@ module.exports = {
   },
   stories: ['../src/**/*.stories.@(tsx|mdx)'],
   addons: [
-    {
-      name: '@storybook/preset-create-react-app',
-      // Override react-scripts package path as it resolves to the wrong path on Linux
-      options: { scriptsPackageName: '../node_modules/react-scripts' },
-    },
+    'storybook-css-modules-preset',
```
Comment (axelboc) [inline apps/storybook/.storybook/main.js:9]: Storybook doesn't support CSS modules by default... :disappointed:

File: apps/storybook/.storybook/preview.js [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import 'normalize.css';
-import '@h5web/lib/src/styles.css';
+import '@h5web/shared/src/styles.css';
```
Comment (axelboc) [inline apps/storybook/.storybook/preview.js]: Storybook doesn't support importing modules with `@import '<module-name>`, so the import of shared package's styles in `lib/src/styles.css` fails. It supports `@import '~<module-name>`, which is webpack's `css-loader` module import syntax, but Vite doesn't support it.

File: apps/storybook/.storybook/preview.js [hunk 3]
```diff
@@ -1,5 +1,5 @@
 import 'normalize.css';
-import '@h5web/lib/src/styles.css';
+import '@h5web/shared/src/styles.css'; // importing via `@h5web/lib/src/styles.css` styles fails due ot lack of `~` prefix (as it is not supported by Vite)
```
Comment (axelboc) [inline apps/storybook/.storybook/preview.js:2]: Storybook doesn't support importing modules with @import '<module-name>, so the import of shared package's styles in lib/src/styles.css fails. It supports @import '~<module-name>, which is webpack's css-loader module import syntax, but Vite doesn't support it.

File: apps/storybook/src/Selection.stories.tsx [hunk 4]
```diff
@@ -36,38 +36,34 @@ const Template: Story<TemplateProps> = (args) => {
...
   return (
-    <>
-      {selection && (
-        <p style={{ textAlign: 'center' }}>
-          {selectedVectors
-            ? `Selection from ${vectorToStr(
-                selectedVectors[0]
-              )} to ${vectorToStr(selectedVectors[1])}`
-            : 'No selection'}
-        </p>
```
Comment (axelboc) [inline apps/storybook/src/Selection.stories.tsx:47]: I noticed that the canvas would grow when opening the story, probably due to `display: grid` on the parent `FillHeight` decorator. So I now pass the current selection via the `title` prop of `VisCanvas`.

File: eslint.shared.js [hunk 5]
```diff
@@ -76,6 +77,7 @@ const overrides = [
       'jest/prefer-strict-equal': 'off', // `toEqual` is shorter and sufficient in most cases
       'jest-formatting/padding-around-all': 'off', // allow writing concise two-line tests
       'jest/require-top-level-describe': 'off', // filename should already be meaningful, extra nesting is unnecessary
+      'jest/no-conditional-in-test': 'off', // false positives in E2E tests (snapshots), and too strict (disallows using `||` for convenience)
```
Comment (axelboc) [inline eslint.shared.js:80]: Disabling some new troublesome linting rules.

**Comment by axelboc** [comment]
I'll rebase and fix the conflicts if you want to merge first :wink:

File: apps/storybook/.storybook/main.js (hunk 1 above)
Comment (loichuder) [reply apps/storybook/.storybook/main.js:9]: Well, it could be worse than just an addon to add :smile:

**Review by loichuder — APPROVED** [review]

**Pull request merged**